lib/srv/ctx.go

```diff
@@ -676,22 +678,82 @@ func (c *ServerContext) String() string {
 	return fmt.Sprintf("ServerContext(%v->%v, user=%v, id=%v)", c.ServerConn.RemoteAddr(), c.ServerConn.LocalAddr(), c.ServerConn.User(), c.id)
 }
 
-// ExecCommand takes a *ServerContext and extracts the parts needed to create
-// an *execCommand which can be re-sent to Teleport.
-func (c *ServerContext) ExecCommand() (*ExecCommand, error) {
-	var pamEnabled bool
-	var pamServiceName string
-	var pamUseAuth bool
+func (c *ServerContext) getPAMConfig() (*PAMConfig, error) {
+	if c.srv.Component() != teleport.ComponentNode {
+		return &PAMConfig{Enabled: false}, nil
+	}
+
+	accessPoint := c.srv.GetAccessPoint()
+	clusterPAMConfig, err := accessPoint.GetPAMConfig(context.TODO())
+
+	// If it doesn't exist, don't return an hard error and fall back to local instead.
+	if err != nil && !trace.IsNotFound(err) {
+		return nil, trace.Wrap(err)
+	}
 
-	// If this code is running on a node, check if PAM is enabled or not.
-	if c.srv.Component() == teleport.ComponentNode {
-		conf, err := c.srv.GetPAM()
+	// Get the local PAM configuration.
+	localPAMConfig, err := c.srv.GetPAM()
+	if err != nil && trace.IsNotFound(err) {
+		return nil, trace.Wrap(err)
```

Comment:
If `err` is not `trace.NotFound` (and also not `nil`), `localPAMConfig` will be `nil` and panic on the next access (`.Override`)